sdk/tests/deqp/functional/gles3/es3fUniformBlockTests.js

```diff
@@ -0,0 +1,753 @@
+        blockA.setFlags(this.m_flagsA);
+
+        /** @type {glsUBC.UniformBlock} */ var blockB = this.m_interface.allocBlock('BlockB');
+        blockB.addUniform(new glsUBC.Uniform('a', glsUBC.newVarTypeBasic(deqpUtils.DataType.FLOAT_MAT2, glsUBC.UniformFlags.PRECISION_MEDIUM)));
+        blockB.addUniform(new glsUBC.Uniform('b', glsUBC.newVarTypeStruct(typeT)));
+        blockB.addUniform(new glsUBC.Uniform('c', glsUBC.newVarTypeBasic(deqpUtils.DataType.BOOL_VEC4, 0), glsUBC.UniformFlags.UNUSED_BOTH));
+        blockB.addUniform(new glsUBC.Uniform('d', glsUBC.newVarTypeBasic(deqpUtils.DataType.BOOL, 0)));
+        blockB.setInstanceName('blockB');
+        blockB.setFlags(this.m_flagsB);
+
+        if (this.m_numInstances > 0)
+        {
+            blockA.setArraySize(this.m_numInstances);
+            blockB.setArraySize(this.m_numInstances);
+        }
+    };
+
+    /**
+     * Creates the test hierarchy to be executed.
+     * @param {string} filter A filter to select particular tests.
+     **/
+    var init = function() {
+        /** @const @type {deqpTests.DeqpTest} */ var testGroup = deqpTests.runner.getState().testCases;
+
+        /** @type {deqpUtils.DataType} */
+        var basicTypes = [
+            deqpUtils.DataType.FLOAT,
+            deqpUtils.DataType.FLOAT_VEC2,
+            deqpUtils.DataType.FLOAT_VEC3,
+            deqpUtils.DataType.FLOAT_VEC4,
+            deqpUtils.DataType.INT,
+            deqpUtils.DataType.INT_VEC2,
+            deqpUtils.DataType.INT_VEC3,
+            deqpUtils.DataType.INT_VEC4,
+            deqpUtils.DataType.UINT,
+            deqpUtils.DataType.UINT_VEC2,
+            deqpUtils.DataType.UINT_VEC3,
+            deqpUtils.DataType.UINT_VEC4,
+            deqpUtils.DataType.BOOL,
+            deqpUtils.DataType.BOOL_VEC2,
+            deqpUtils.DataType.BOOL_VEC3,
+            deqpUtils.DataType.BOOL_VEC4,
+            deqpUtils.DataType.FLOAT_MAT2,
+            deqpUtils.DataType.FLOAT_MAT3,
+            deqpUtils.DataType.FLOAT_MAT4,
+            deqpUtils.DataType.FLOAT_MAT2X3,
+            deqpUtils.DataType.FLOAT_MAT2X4,
+            deqpUtils.DataType.FLOAT_MAT3X2,
+            deqpUtils.DataType.FLOAT_MAT3X4,
+            deqpUtils.DataType.FLOAT_MAT4X2,
+            deqpUtils.DataType.FLOAT_MAT4X3
+        ];
+
+        /** @type {Array.<string, glsUBC.UniformFlags>} */
+        var precisionFlags = [
+            { name: 'lowp', flags: glsUBC.UniformFlags.PRECISION_LOW },
+            { name: 'mediump', flags: glsUBC.UniformFlags.PRECISION_MEDIUM },
+            { name: 'highp', flags: glsUBC.UniformFlags.PRECISION_HIGH }
+        ];
+
+        /** @type {Array.<string, glsUBC.UniformFlags>} */
+        var layoutFlags = [
+            { name: 'shared', flags: glsUBC.UniformFlags.PRECISION_LOW },
+            { name: 'packed', flags: glsUBC.UniformFlags.PRECISION_MEDIUM },
+            { name: 'std140', flags: glsUBC.UniformFlags.PRECISION_HIGH }
```

Comment:
Also, see https://github.com/KhronosGroup/WebGL/issues/911 -- some time ago the plan was to remove the 'packed' layout qualifier since it could cause portability issues for content.